open-craft/openedx-completion-aggregator
# Taniwha/bb 2497/ironwood fix root block

**Description:** Handle invalid key errors from opaque_keys, treating them as 'not found' 

**JIRA:** https://tasks.opencraft.com/browse/BB-2497

**Dependencies:** no dependencies

**Merge deadline:** List merge deadline (if any)

**Installation instructions:** nothing special

**Testing instructions:**

1. Set up an ironwood devstack (I had a lot of trouble with this, check jira ticket for details)
2. Install the completion aggregator in lms (`make lms-shell`, `pip install /etc/src/openedx-completion-aggregator`, `make lms-restart`)
3. Log into the lms site (http://localhost:18000)
4. Go to http://localhost:18000/completion-aggregator/v1/course/course-v1:edX+DemoX+Demo_Course/ You may get a programming error
    ```
    Exception Type: ProgrammingError at /completion-aggregator/v1/course/course-v1:edX+DemoX+Demo_Course/
    Exception Value: (1146, "Table 'edxapp.completion_aggregator_aggregator' doesn't exist")
    ```
    However, the exception comes after the problem line.
5. Enter a URL with an invalid key, eg; http://localhost:18000/completion-aggregator/v1/course/course-v1:edX+DemoX-Demo_Course/
6. You should see a "Not Found" message:
    ![not_found](https://user-images.githubusercontent.com/1664867/83846355-6c503800-a745-11ea-8525-550c7d146d99.png)


**Reviewers:**

**Merge checklist:**

**Post merge:**
      finished.

**Author concerns:** no known concerns
